eclipse-collections-code-generator/src/main/resources/impl/list/immutable/primitiveSingletonSpliterator.stg

```diff
@@ -0,0 +1,42 @@
+primitiveSingletonSpliterator(name, type) ::= <<
+private static final class SingletonSpliterator implements Spliterator.Of<name>
```

Comment:
Sorry, I didn't mean just a different stg file to be included. I meant a public class that can be used without having to have this inner class in multiple places. For example, if we implement streams for `ImmutableDoubleSingletonSet`, it would be good to use the same class. The list package is not the right place for that impl. I think the `streams` package could use a `primitive` subpackage to house this.
@motlin what do you think?

Reply:
I will need some help here as i just create these spliterators in case of int,long and double. How do i create these classes from an stg file on conditinal base? I know how to include parts of files in terms of <if(primitive.specializedStream)> but how do i create classes just if <if(primitive.specializedStream)>?

Reply:
I found a way. i can change EclipseCollectionsCodeGenerator and set a flag in the stg file. Is this ok? @motlin would like to have your feedback here

Reply:
i spliited this into another stg file which now generates classes under the stream directory.
I think this is the right place because the existing class there deals with streams.

Reply:
@mohrezaei please check if this is what you wanted.

Reply:
> I found a way. i can change EclipseCollectionsCodeGenerator and set a flag in the stg file. Is this ok? @motlin would like to have your feedback here

@david-gang this is fine to add that check in the Code Generator